SConstruct

```diff
@@ -1338,6 +1338,23 @@ env.AlwaysBuild(
             env.PrettyCommand('TIDY', 'src', 'yellow')
         )))
 
+if 'PKG_CONFIG_PATH' in env.Dictionary():
+    pkg_config_path = env['PKG_CONFIG_PATH']
+elif 'PKG_CONFIG' in env.Dictionary() or env.Which('pkg-config'):
+    # https://linux.die.net/man/1/pkg-config the default is libdir/pkgconfig
+    pkg_config_path = os.path.join(env['ROC_SYSTEM_LIBDIR'], 'pkgconfig')
+else:
+    pkg_config_path = None
+if pkg_config_path:
+    pc_target = env.PkgConfig(
+                    build_dir='build',
+                    prefix=GetOption('prefix'),
+                    libdir=env['ROC_SYSTEM_LIBDIR'],
+                    filename='libroc.pc',
+                    version=env['ROC_VERSION'],
+                    dependencies=system_dependencies)
+    env.AddDistFile(pkg_config_path, pc_target)
```

Comment:
Let's move this call to `src/SConscript`. `SConstruct` file is mostly for configuration, and `SConscript` file is for performing real actions.

We should generate pkg-config only if building libroc was not diabled. In SConscript, we already have this section:

```
if not GetOption('disable_lib'):
    ...
```

So let's move this piece to the end of that section.